src/search.cpp

```diff
@@ -421,9 +427,23 @@ void Search::Worker::iterative_deepening() {
         if (!threads.stop)
             completedDepth = rootDepth;
 
-        if (rootMoves[0].pv[0] != lastBestMove)
+        // We make sure not to pick an unproven mated-in score,
+        // in case this thread prematurely stopped search (aborted-search).
+        if (threads.abortedSearch && rootMoves[0].score != -VALUE_INFINITE
+            && rootMoves[0].score <= VALUE_TB_LOSS_IN_MAX_PLY)
         {
-            lastBestMove      = rootMoves[0].pv[0];
+            // Bring the last best move to the front for best thread selection.
+            std::sort(rootMoves.begin(), rootMoves.end(),
+                      [lastBestPV](const RootMove& rootMove, const RootMove&) {
```

Comment:
all is well, though we might want to avoid the vector copy, so we can rewrite that as follows, while also keeping `lastBestPV` immutable
```cpp
[&lastBestPV = std::as_const(lastBestPV)](const RootMove& rootMove, const RootMove&){
//
}
```

needs the header `utility` if not already present